eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/persistence/DbsnpClusteredVariantEntity.java

```diff
@@ -0,0 +1,141 @@
+
+import uk.ac.ebi.eva.accession.core.ClusteredVariant;
+import uk.ac.ebi.eva.accession.core.IClusteredVariant;
+import uk.ac.ebi.eva.accession.core.VariantType;
+
+import java.util.Arrays;
+
+@Document
+public class DbsnpClusteredVariantEntity extends AccessionedDocument<Long> implements IClusteredVariant {
+
+    @Field("asm")
+    private String assemblyAccession;
+
+    @Field("tax")
+    private int taxonomyAccession;
+
+    private String contig;
+
+    private long start;
+
+    private VariantType type;
+
+    @Field("validated")
+    private Boolean validated;
+
+    protected DbsnpClusteredVariantEntity() {
+    }
+
+    public DbsnpClusteredVariantEntity(Long accession, String hashedMessage, IClusteredVariant model) {
+        this(accession, hashedMessage, model.getAssemblyAccession(), model.getTaxonomyAccession(), model.getContig(),
+             model.getStart(), model.getType(), model.isValidated(), 1);
+    }
+
+    public DbsnpClusteredVariantEntity(Long accession, String hashedMessage, String assemblyAccession,
+                                       int taxonomyAccession, String contig, long start, VariantType type,
+                                       Boolean validated, int version) {
+        super(hashedMessage, accession, version);
+        this.assemblyAccession = assemblyAccession;
+        this.taxonomyAccession = taxonomyAccession;
+        this.contig = contig;
+        this.start = start;
+        this.type = type;
+        this.validated = validated;
+    }
+
+    public IClusteredVariant getModel() {
+        return new ClusteredVariant(this);
+    }
+
+    @Override
+    public String getAssemblyAccession() {
+        return assemblyAccession;
+    }
+
+    @Override
+    public int getTaxonomyAccession() {
+        return taxonomyAccession;
+    }
+
+    @Override
+    public String getContig() {
+        return contig;
+    }
+
+    @Override
+    public long getStart() {
+        return start;
+    }
+
+    @Override
+    public VariantType getType() {
+        return type;
+    }
+
+    @Override
+    public Boolean isValidated() {
+        return validated;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof DbsnpClusteredVariantEntity)) {
+            return false;
+        }
+
+        DbsnpClusteredVariantEntity that = (DbsnpClusteredVariantEntity) o;
+
+        if (taxonomyAccession != that.taxonomyAccession) {
+            return false;
+        }
+        if (start != that.start) {
+            return false;
+        }
+        if (!assemblyAccession.equals(that.assemblyAccession)) {
+            return false;
+        }
+        if (!contig.equals(that.contig)) {
+            return false;
+        }
+        if (type != that.type) {
+            return false;
+        }
+        return validated != null ? validated.equals(that.validated) : that.validated == null;
```

Comment:
Again, validated is not in the summary function and maybe it should not be in the equals and hashCode.

Reply:
> we agreed to use equals as just java-wise (for tests about all the fields and such). we have the summary function to work with the (bussiness logic of) identifying fields